vaultManager/vaultManager.go

```diff
@@ -63,34 +99,58 @@ func baseVaultPath(dataType string) string {
 	return fmt.Sprintf("%s/%s/%s", remote_signer.VaultBackend, dataType, remote_signer.VaultNamespace)
 }
 
-func (vm *VaultManager) getClient() *api.Client {
+func (vm *VaultManager) validTokenTTL() bool {
+	if vm.token.getTime != nil {
+		var now = time.Now().Unix()
+		var timeWithTTL = vm.token.getTime.Unix() + int64(vm.token.ttl.Seconds())
 
-	options := map[string]interface{}{
-		"token": vm.client.Token(),
+		if now >= timeWithTTL {
```

Comment:
This line could be reduced to a return statement using the [After](https://golang.org/pkg/time/#Time.After) method of the time package for indicating if the `timeWithTTL` is expired.
```suggestion
		return timeWithTTL.After(now)
```